Server/GameServer/User.cpp

```diff
@@ -1197,40 +1199,64 @@ void CUser::UpdateFogDensity(Packet & pkt)
 void CUser::SendTime()
 {
 	Packet result(WIZ_TIME);
-	result	<< uint16_t(g_pMain->m_sYear) << uint16_t(g_pMain->m_sMonth) << uint16_t(g_pMain->m_sDate)
-		<< uint16_t(g_pMain->m_sHour) << uint16_t(g_pMain->m_sMin);
+	result	<< uint16_t(g_pMain->m_sYear) << uint16_t(g_pMain->m_sMonth) << uint16_t(g_pMain->m_sDay)
+		    << uint16_t(g_pMain->m_sHour) << uint16_t(g_pMain->m_sMin);
 	Send(&result);
 }
 
 /**
-* @brief Overloads CUser::SendTime() to change the server time by GM command.
+* @brief	Overloads CUser::SendTime() to change the server time by GM command.
+*			NOTE: The client can support date, but currenty commented out in CGameProcMain::ParseChattingCommand->CMD_TIME.
+*			So if you need it to support for whatever reason, just remove the comment from:
+*			std::string sDate = szCmds[2];
+* 
+* @param	pkt	The packet.
 */
 void CUser::SendTime(Packet & pkt)
 {
 	if (!this->isGM()) return;
+ 
+	uint16_t iYear  = pkt.read<uint16_t>();
+	uint16_t iMonth = pkt.read<uint16_t>();
+	uint16_t iDay   = pkt.read<uint16_t>();
+	uint16_t iHour  = pkt.read<uint16_t>();
+	uint16_t iMin   = pkt.read<uint16_t>();
+
+	if (iHour == 0)
```

Comment:
I think your problem here is trying to enforce your own logic/behaviour to an existing, official command/packet.

Rather than trying to workaround an issue created by trying to enforce this custom logic, I'd suggest going back and seeing how it works in the official 1.298 client, and then having it work the same here.

This is why I'm very much against custom behaviour. It's overcomplicating what should be trivial, and causing confusion where it doesn't need to exist.